flink-libraries/flink-table/src/test/scala/org/apache/flink/table/api/scala/stream/sql/SqlITCase.scala

```diff
@@ -317,4 +397,24 @@ class SqlITCase extends StreamingWithStateTestBase {
     result.addSink(new StreamITCase.StringSink)
     env.execute()
   }
+
+}
+
+object SqlITCase {
+
+  class TimestampWithLatenessWatermark(allowedLateness: Long)
```

Comment:
I don't think we need this timestamp extractor. IMO, the offset should be integrated into the window operator itself and not rely on modified watermarks. 
I'd suggest to first add this OVER window operator without offset support and add this later when the other OVER window operators have been added. 

What do you think, @shaoxuan-wang and @sunjincheng121 ?